specification/programmableconnectivity/ProgrammableConnectivity.Management/Gateway.tsp

```diff
@@ -51,6 +51,231 @@ model GatewayProperties {
    */
   @visibility("read")
   provisioningState?: ProvisioningState;
+
+  /**
+   * Details about the Application that would use the Operator's Network APIs.
+   */
+  @TypeSpec.Versioning.added(Versions.v2024_11_01_preview)
+  @visibility("read", "create", "update")
+  configuredApplication: ApplicationProperties;
+
+  /**
+   * Details about the Organization owning the Application that would use the Operator's Network APIs.
+   */
+  @TypeSpec.Versioning.added(Versions.v2024_11_01_preview)
+  @visibility("read", "create", "update")
+  configuredApplicationOwner: ApplicationOwnerProperties;
+}
+
+/**
+ * Details about the Application Owner (i.e.e organization) that would use the Operator's Network APIs.
+ */
+@TypeSpec.Versioning.added(Versions.v2024_11_01_preview)
+model ApplicationOwnerProperties {
+  /**
+   * Common name of the organization owning the application.
+   */
+  name: string;
+
+  /**
+   * Legal name of the organization owning the application.
+   */
+  legalName: string;
+
+  /**
+   * Trading name of the organization owning the application.
+   */
+  tradingName: string;
+
+  /**
+   * A description of the organization owning the application.
+   */
+  organizationDescription: string;
+
+  /**
+   * Unique Tax Number for the user's organization in the country/region the APC Gateway is being purchased.
+   */
+  taxNumber: string;
+
+  /**
+   * ID number for official identification of the organization
+   */
+  organizationIdentificationId: string;
+
+  /**
+   * Issuing organization of the ID for official identification of the organization
+   */
+  organizationIdentificationIssuer: string;
+
+  /**
+   * Contact email address for the use of Network APIs at the company
+   */
+  contactEmailAddress: string;
+
+  /**
+   * Legal representative of the organization
+   */
+  legalRepresentative: Person;
+
+  /**
+   * Privacy manager of the organization
+   */
+  privacyManager: Person;
+
+  /**
+   * Registered Geographic address
+   */
+  registeredGeographicAddress: GeographicAddress;
+}
+
+/**
+ * Details about the Application that would use the Operator's Network APIs.
+ */
+model ApplicationProperties {
+  /**
+   * Name of the application. Example: Contoso App.
+   */
+  @TypeSpec.Versioning.madeRequired(Versions.v2024_11_01_preview)
+  name: string;
+
+  /**
+   * Description of the application.
+   */
+  @TypeSpec.Versioning.madeRequired(Versions.v2024_11_01_preview)
+  applicationDescription: string;
+
+  /**
+   * The category that describes the application.
+   */
+  @TypeSpec.Versioning.removed(Versions.v2024_11_01_preview)
```

Comment:
I don't think we need to do a diff between this object (configuredApplication in a Gateway) and the pre-existing one (configuredApplication in an OAC). Gateway and OperatorApiConnection are distinct ARM resources, so I think all we need to do is say "we've removed a property called configuredApplication from OAC, and separately we've added a property called configuredApplication to Gateway" - there's no expectation that these two fields would be the same object

Reply:
The openapi.json for the previous version doesn't generate correctly if I don't do this - I think for TypeSpec, the .tsp file the ApplicationProperties Model is in doesn't matter, it generates the same model in the openapi.json whichever file it's in and for back-compatibility reasons we can't change that. I could have deprecated the entire previous model and added a new model with a new name, but I'd prefer to keep the same model names